src/Avalonia.Visuals/Media/Color.cs

```diff
@@ -14,6 +14,11 @@ namespace Avalonia.Media
 #endif
     readonly struct Color : IEquatable<Color>
     {
+        /// <summary>
+        /// Represents a null <see cref="Color"/> with zero set for all channels.
+        /// </summary>
+        public static readonly Color Empty = new Color();
```

Comment:
I was surprised this didn't already exist in Avalonia. It is the same as WPF now.

https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color.empty?view=net-6.0

Reply:
Marking resolved since no additional comments in review

Reply:
I was actually surprised that this is a thing in WPF, since from my perspective "empty" color makes no sense semantically and I wonder how this is intended to be used. But then I checked WPF color and it has no such weird members https://docs.microsoft.com/en-us/dotnet/api/system.windows.media.color?view=windowsdesktop-6.0.

System.Drawing docs about "Represents a color that is null." are a total gibberish to me and looks like a poor man's nullable color.

I vote against having this (and `IsEmpty`) in the public API since it is just confusing. @AvaloniaUI/core

Reply:
I've needed this in the past. A good example is a color picker where it's necessary to know if the user made a selection or not. This also exists in the System.Drawing.Color class.

Reply:
`Nullable<HsvColor>` is similar as you pointed out; however, there is no need to have such overhead when the color struct itself can represent such a case. It cleans up a lot of API's that would otherwise need a nullable color to communicate such information.

Reply:
Just did a search for "Color.IsEmpty" online as well. A LOT of implementations do this as well.

Reply:
I mostly checked modern frameworks and didn't find it anywhere:
- https://api.flutter.dev/flutter/dart-ui/Color-class.html
- https://developer.apple.com/documentation/swiftui/color
- https://doc.qt.io/qt-5/qcolor.html

Do you have an example from other framework? I would be curious to see their reasoning.

Reply:
This is still current in .NET. I didn't check other frameworks.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Drawing.Primitives/src/System/Drawing/Color.cs

Note that `IsEmpty`; however, is tracked with a separate "state" field. That would mitigate your concern about it being equal to #00000000 which is technically a valid color.

My understanding was in older .NET Framework it was implemented as I did here. However, that may be wrong:
https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Color.cs,1433

Here are some .NET examples:

https://docs.microsoft.com/en-us/dotnet/api/system.drawing.color.empty?view=net-6.0
https://help.syncfusion.com/cr/file-formats/Syncfusion.Presentation.IColor.html#Syncfusion_Presentation_IColor_IsEmpty
https://www.grapecity.com/activereportsnet/docs/v14/online/GrapeCity.ActiveReports.Viewer.Wpf~GrapeCity.Viewer.Common.Color~Empty.html

It was even done for other types:

https://docs.microsoft.com/en-us/dotnet/api/system.drawing.point.isempty?view=net-6.0
https://docs.microsoft.com/en-us/dotnet/api/system.drawing.rectangle.empty?view=net-6.0

Reply:
If you need more opinions, I also don't see much sense in Empty/IsEmpty here. 
For "#00000000" state we already have Transparent value.

It would make more meaning if we had some special flag as in winforms, but we don't. Also I am not sure if we should have it, might it affect interoperability? In any way special flag discussion (as well as adding IsEmpty after that) is out of this PR imo.

Reply:
> For "#00000000" state we already have Transparent value

These are not the same as mentioned above. #00000000 != #00FFFFFF. Transparent was intentionally set to this value for this reason.

Aside from the misunderstanding with the field, I certainly think this is still useful. You guys are really hard to convince sometimes though :) I'm not inventing anything new here.

Reply:
Moreover, do we **really** need that kind of information in Color struct? Where can it be used, where Transparent or nullable value cannot be used?

Reply:
> #00000000 != #00FFFFFF

That's another confusing moment. Understanding of what is transparent isn't consistent across frameworks. General understanding is both (or any color with 0 alpha channel).

In wpf it's #00ffffff, when in cocoa it's called "clear"  and has value of #00000000, and in html5 there is "transparent" with value of "rgba(0,0,0,0)".

Reply:
> Understanding of what is transparent isn't consistent across frameworks.

Point taken. But transparency isn't really what this is about.

All platforms should render #00FFFFFF as transparent too. Obviously, you don't translate `KnownColors.Transparent` for each backend. You have it defined once with #00FFFFFF as in .NET and it works well enough.

Concerns about different interpretations of empty is solvable by adding the `int state` field to track initialization and empty status. Standardizing what an Empty color means across platforms might actually be an argument in favor of keeping this functionality.

Reply:
Sorry but i'll have to agree to the objections that @MarchingCube @maxkatz6 on the same grounds as well.

Reply:
Alright, that is 3 against 1 and no one other than myself is defending this -- which surprises me a little because again this isn't a new idea and there is past precedent. However, since it isn't required functionality right now I will remove it from HsvColor and Color and then create a separate issue to discuss and perhaps get wider attention and feedback over time.

I would prefer not to have a nullable ColorPicker in the future when this could all be managed in the Color structs themselves. I know performance and GDI+ reasons were a consideration when this was originally done as well.

Reply:
I can live with `Color.IsDefault`

Reply:
Color.IsDefault is not a standardized API... but it would convey the same purpose.

Reply:
I created a new issue: https://github.com/AvaloniaUI/Avalonia/issues/7762.

This is resolved but I'm leaving it unresolved so the links to this discussion work in the future.